src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -554,6 +555,11 @@ public override Task FinishAsync(CancellationToken cancellationToken)
             _readLength = 0;
             _readOffset = 0;
 
+            // Issue a pre-emptive read on the stream, to allow us to determine
+            // whether the pooled connection has been closed and whether any
+            // erroneous data has been written to it.
+            _initialRead = _stream.ReadAsync(_readBuffer, 0, _readBuffer.Length);
```

Comment:
This isn't strictly necessary, is it?  If so, the comment is misleading.  It certainly does hurt to do this, but is it the right thing?